input/size_test.go

```diff
@@ -281,10 +307,533 @@ func TestTxWeightEstimator(t *testing.T) {
+				KeyDesc: keychain.KeyDescriptor{
+					PubKey: testPubkey,
+				},
+				DoubleTweak: testPrivkey,
+			}
+
+			witness, err := input.ReceiverHtlcSpendRevoke(
+				&dummySigner{}, signDesc, testTx,
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			return witness
+		},
+	},
+	{
+		name:    "accepted htlc timeout",
+		expSize: input.AcceptedHtlcTimeoutWitnessSize - 3,
+		genWitness: func(t *testing.T) wire.TxWitness {
+
+			witScript, err := input.ReceiverHTLCScript(
+				testCLTVExpiry, testPubkey, testPubkey,
+				testPubkey, testHash160, false,
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			signDesc := &input.SignDescriptor{
+				WitnessScript: witScript,
+			}
+
+			witness, err := input.ReceiverHtlcSpendTimeout(
+				&dummySigner{}, signDesc, testTx,
+				testCLTVExpiry,
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			return witness
+		},
+	},
+	{
+		name:    "accepted htlc timeout confirmed",
+		expSize: input.AcceptedHtlcTimeoutWitnessSize,
+		genWitness: func(t *testing.T) wire.TxWitness {
+			witScript, err := input.ReceiverHTLCScript(
+				testCLTVExpiry, testPubkey, testPubkey,
+				testPubkey, testHash160, true,
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			signDesc := &input.SignDescriptor{
+				WitnessScript: witScript,
+			}
+
+			witness, err := input.ReceiverHtlcSpendTimeout(
+				&dummySigner{}, signDesc, testTx,
+				testCLTVExpiry,
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			return witness
+		},
+	},
+	{
+		name:    "accepted htlc success",
+		expSize: input.AcceptedHtlcSuccessWitnessSize - 3,
+		genWitness: func(t *testing.T) wire.TxWitness {
+			witScript, err := input.ReceiverHTLCScript(
+				testCLTVExpiry, testPubkey, testPubkey,
+				testPubkey, testHash160, false,
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			signDesc := &input.SignDescriptor{
+				WitnessScript: witScript,
+				KeyDesc: keychain.KeyDescriptor{
+					PubKey: testPubkey,
+				},
+			}
+
+			witness, err := input.ReceiverHtlcSpendRedeem(
+				&maxDERSignature{}, txscript.SigHashAll,
+				testPreimage, &dummySigner{}, signDesc, testTx,
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			return witness
+		},
+	},
+	{
+		name:    "accepted htlc success confirmed",
+		expSize: input.AcceptedHtlcSuccessWitnessSize,
+		genWitness: func(t *testing.T) wire.TxWitness {
+			witScript, err := input.ReceiverHTLCScript(
+				testCLTVExpiry, testPubkey, testPubkey,
+				testPubkey, testHash160, true,
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			signDesc := &input.SignDescriptor{
+				WitnessScript: witScript,
+				KeyDesc: keychain.KeyDescriptor{
+					PubKey: testPubkey,
+				},
+			}
+
+			witness, err := input.ReceiverHtlcSpendRedeem(
+				&maxDERSignature{}, txscript.SigHashAll,
+				testPreimage, &dummySigner{}, signDesc, testTx,
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			return witness
+		},
+	},
+}
+
+// TestWitnessSizes asserts the correctness of our magic witness constants.
```

Comment:
Solid test 👍 

This gets us closer to just eventually using the raw stripped size in our core logic rather than relying on these estimates (but ofc the fixed issues with the _witness_ sizes still apply).